Resources/ProxyProvider/PUSPProxyProvider.py

```diff
@@ -18,49 +18,35 @@ def __init__(self, parameters=None):
 
     super(PUSPProxyProvider, self).__init__(parameters)
 
-  def getProxy(self, userDict):
-    """ Generate user proxy
+  def checkStatus(self, userDN):
+    """ Read ready to work status of proxy provider
 
-        :param dict userDict: user description dictionary with possible fields:
-               FullName, UserName, DN, Email, DiracGroup
+        :param str userDN: user DN
 
-        :return: S_OK(basestring)/S_ERROR() -- basestring is a proxy string
+        :return: S_OK(dict)/S_ERROR() -- dictionary contain fields:
+                  - 'Status' with ready to work status[ready, needToAuth]
     """
+    if not userDN.split(":")[-1]:
```

Comment:
For my understanding, what are you looking for here?

Reply:
Here I am trying to find a unique identifier (username) in this DN to use as `cn-label` to generate PUSP. If it's missing, it must be fail.

Reply:
Can you add it it as comment?

Reply:
ok